src/dicg.jl

```diff
@@ -491,52 +515,86 @@ function lazy_dicg_step(
 )
     v_local, v_local_loc, val, a_local, a_local_loc, valM =
-        pre_computed_set_argminmax(pre_computed_set, gradient)
-    step_type = ST_REGULAR
+        pre_computed_set_argminmax(lmo, pre_computed_set, gradient, x; strong_lazification = strong_lazification)
+    step_type = ST_PAIRWISE
     away_index = nothing
     fw_index = nothing
     grad_dot_x = fast_dot(x, gradient)
     grad_dot_a_local = valM
-
-    # Do lazy pairwise step
     grad_dot_lazy_fw_vertex = val
 
-    if grad_dot_a_local - grad_dot_lazy_fw_vertex >= phi / lazy_tolerance &&
-       grad_dot_a_local - grad_dot_lazy_fw_vertex >= epsilon
+    if strong_lazification
+        a_taken = a_local
+        grad_dot_a_taken = grad_dot_a_local
+    else
+        a_taken = compute_inface_extreme_point(lmo, NegatingArray(gradient), x)
+        grad_dot_a_taken = fast_dot(gradient, a_taken)
+    end
+    # Do lazy pairwise step
+    if grad_dot_a_taken - grad_dot_lazy_fw_vertex >= phi / lazy_tolerance &&
+       grad_dot_a_taken - grad_dot_lazy_fw_vertex >= epsilon
         step_type = ST_LAZY
         v = v_local
-        a = a_local
+        a = a_taken
         d = muladd_memory_mode(memory_mode, d, a, v)
         fw_index = v_local_loc
     else
         v = compute_extreme_point(lmo, gradient)
         grad_dot_v = fast_dot(gradient, v)
-        # Do lazy inface_point
-        if grad_dot_a_local - grad_dot_v >= phi / lazy_tolerance && 
-            grad_dot_a_local - grad_dot_v >= epsilon
-            step_type = ST_LAZY
-            a = a_local
-            away_index = a_local_loc
+        dual_gap = grad_dot_x - grad_dot_v
+        phi = dual_gap
+
+        if variant == "standard"
+            v_taken = v
+            grad_dot_v_taken = grad_dot_v
         else
-            a = compute_inface_extreme_point(lmo, NegatingArray(gradient), x)
+            v_taken = compute_inface_extreme_point(lmo, gradient, x;)
+            grad_dot_v_taken = fast_dot(gradient, v_taken)
+
         end
-        
-        # Real dual gap promises enough progress.
-        grad_dot_fw_vertex = fast_dot(v, gradient)
-        dual_gap = grad_dot_x - grad_dot_fw_vertex
-        
+        # Do lazy inface_point
         if dual_gap >= phi / lazy_tolerance
-            d = muladd_memory_mode(memory_mode, d, a, v)
-            #Lower our expectation for progress.
+            if (grad_dot_a_taken - grad_dot_v_taken >= phi / lazy_tolerance &&
+                grad_dot_a_taken - grad_dot_v_taken >= epsilon) || !strong_lazification
+                if !strong_lazification
+                    a = a_taken
+                    d = muladd_memory_mode(memory_mode, d, a, v_taken)
+                    step_type = ST_PAIRWISE
+                    away_index = -1
+                else
+                    a = a_taken
+                    d = muladd_memory_mode(memory_mode, d, a, v_taken)
+                    step_type = ST_PAIRWISE
+                    away_index = a_local_loc
+                end
```

Comment:
The only thing that is different is `away_index`, isn't it? That can be done cleaner as
```
away_index = strong_lazification ? a_local_loc : -1
```